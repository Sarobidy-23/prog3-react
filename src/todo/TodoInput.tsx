import React, { useState } from 'react'

interface Props {
    insert: (name: string)=>void
}

function TodoInput(props: Props) {
    const {insert} = props;
    const [todoName, setTodoName] = useState<string>("")

    return (
        <input type="text"
            className='input-new_todo'
            placeholder='new todo name'
            value={todoName}
            onChange={(e)=>{setTodoName(e.target.value);}}
            onKeyDown={(e)=>{e.key ==='Enter' && insert(todoName)}}/>
    )
}

export default TodoInput
