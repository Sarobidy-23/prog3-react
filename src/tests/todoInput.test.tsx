import renderer, { ReactTestInstance, ReactTestRendererJSON } from 'react-test-renderer';
import { act } from 'react-test-renderer';

import TodoInput from '../todo/TodoInput';

it("Snapshot: render TodoInput component", () => {
    const tree = renderer.create(
        <TodoInput insert={jest.fn}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

it("Verify input placeholder", () => {
    const tree = renderer.create(
        <TodoInput insert={jest.fn}/>
    ).toJSON() as ReactTestRendererJSON;
    expect(tree.props.placeholder).toBe("new todo name")
})

it("Verify onChange in the input", () => {
    const tree = renderer.create(<TodoInput insert={jest.fn()}/>);
    const instance = tree.root as ReactTestInstance;
    
    // eslint-disable-next-line testing-library/await-async-query
    let input = instance.findByProps({placeholder: "new todo name"});
    act(()=>{
        input.props.onChange({target: {value: "test"}})
    })

    expect(input.props.value).toBe("test")
})