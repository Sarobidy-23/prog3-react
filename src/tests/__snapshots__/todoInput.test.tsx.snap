// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: render TodoInput component 1`] = `
<input
  className="input-new_todo"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="new todo name"
  type="text"
  value=""
/>
`;
